### PR TITLE
feat: Update patient info

### PR DESCRIPTION
**Pull request by sgoycoechea-lightit**
# Light It Pull Request Template ⚡⚡⚡

_Resolve_: [closes -#865ca2pnd](https://app.clickup.com/t/865ca2pnd)

## Description ⭐

Implemented the update patient info endpoint, creating the user roles patient and doctor

### Type of change 🏃

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### Requires ❗

- [x] This change requires a database update or migration
- [ ] This change requires a documentation update

## How Has This Been Tested? ❓

Created the following tests:
- Test that a patients info can be correctly updated
- Test that the info can't be updated if the logged user is a doctor
- Test that the info can't be updated if there is no logged in user
- Test that the info can't be updated if there are fields missing

## Checklist ✅

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: database/factories/UserFactory.php [hunk 1]
```diff
@@ -21,11 +26,42 @@ public function definition(): array
             'name' => fake()->name(),
             'email' => fake()->unique()->safeEmail(),
             'email_verified_at' => now(),
-            'password' => '$2y$10$92IXUNpkjO0rOQ5byMi.Ye4oKoEa3Ro9llC/.og/at2.uheWG/igi', // password
+            'password' => Hash::make('password'),
             'remember_token' => Str::random(10),
         ];
     }
 
+    public function doctor()
```
Comment (sojeda) [inline database/factories/UserFactory.php]: return type

File: database/factories/UserFactory.php [hunk 2]
```diff
@@ -21,11 +26,42 @@ public function definition(): array
...
+            Role::findOrCreate(UserRole::DOCTOR->value);
+            $user->assignRole(UserRole::DOCTOR->value);
+        });
+    }
+
+    public function patient()
+    {
+        return $this->afterCreating(function (User $user) {
+            $patientRole = Role::findOrCreate(UserRole::PATIENT->value);
+            Permission::findOrCreate('update personal information');
```
Comment (sojeda) [inline database/factories/UserFactory.php:46]: return type

File: database/factories/UserFactory.php [hunk 3]
```diff
@@ -21,11 +26,42 @@ public function definition(): array
...
+            $patientRole->givePermissionTo('update personal information');
+            $user->assignRole(UserRole::PATIENT->value);
+        });
+    }
+
+    public function withInformation()
```
Comment (sojeda) [inline database/factories/UserFactory.php]: return type

File: database/migrations/2014_10_12_000000_create_users_table.php [hunk 4]
```diff
@@ -17,6 +17,10 @@ public function up(): void
             $table->string('email')->unique();
             $table->timestamp('email_verified_at')->nullable();
             $table->string('password');
+            $table->string('phone_number')->nullable();
+            $table->decimal('weight', 8, 2)->nullable();
+            $table->decimal('height', 8, 2)->nullable();
+            $table->text('other_information')->nullable();
```
Comment (sojeda) [inline database/migrations/2014_10_12_000000_create_users_table.php]: I don't know if I really like leaving null fields for the doctor

File: app/Transformers/UserTransformer.php [hunk 5]
```diff
@@ -31,10 +32,41 @@ class UserTransformer extends Transformer
...
+        $role = $this->getUserRole($user);
+        $data = [
             'id' => (int) $user->id,
             'name' => (string) $user->name,
             'email' => (string) $user->email,
+            'role' => (string) $role,
         ];
+
+        if ($role === UserRole::PATIENT->value) {
+            if ($user->phone_number) {
```
Comment (sojeda) [inline app/Transformers/UserTransformer.php]: Why the if?  You can send the keys with null value

File: database/migrations/2014_10_12_000000_create_users_table.php (hunk 4 above)
Comment (sgoycoechea-lightit) [reply database/migrations/2014_10_12_000000_create_users_table.php]: So you would separate the users in two different tables? Or how would you approach it?

Comment (sojeda) [reply database/migrations/2014_10_12_000000_create_users_table.php]: Its not necesary, you can use user table for primary and another table with patient information

**Review by sojeda — APPROVED** [review]

**Pull request merged**